src/clap/fmt_span.rs

```diff
@@ -0,0 +1,62 @@
+//! Adapter for pasing the [`FmtSpan`] type.
+
+use clap::builder::EnumValueParser;
+use clap::builder::PossibleValue;
+use clap::builder::TypedValueParser;
+use clap::builder::ValueParserFactory;
+use tracing_subscriber::fmt::format::FmtSpan;
+
+/// Wrapper around [`FmtSpan`].
+#[derive(Clone)]
+pub struct FmtSpanWrapper(FmtSpan);
+
+impl From<FmtSpanWrapper> for FmtSpan {
+    fn from(value: FmtSpanWrapper) -> Self {
+        value.0
+    }
+}
+
+impl clap::ValueEnum for FmtSpanWrapper {
+    fn value_variants<'a>() -> &'a [Self] {
+        &[
+            Self(FmtSpan::NEW),
+            Self(FmtSpan::ENTER),
+            Self(FmtSpan::EXIT),
+            Self(FmtSpan::CLOSE),
+            Self(FmtSpan::NONE),
+            Self(FmtSpan::ACTIVE),
+            Self(FmtSpan::FULL),
+        ]
+    }
+
+    fn to_possible_value(&self) -> Option<PossibleValue> {
+        Some(match self.0 {
+            FmtSpan::NEW => PossibleValue::new("new").help("Log when spans are created"),
+            FmtSpan::ENTER => PossibleValue::new("enter").help("Log when spans are entered"),
+            FmtSpan::EXIT => PossibleValue::new("exit").help("Log when spans are exited"),
+            FmtSpan::CLOSE => PossibleValue::new("close").help("Log when spans are dropped"),
+            FmtSpan::NONE => PossibleValue::new("none").help("Do not log span events"),
+            FmtSpan::ACTIVE => {
+                PossibleValue::new("active").help("Log when spans are entered/exited")
+            }
+            FmtSpan::FULL => PossibleValue::new("full").help("Log all span events"),
+            _ => {
+                return None;
+            }
+        })
```

Comment:
Is this one of those permanently non-exhaustive enum things, or are you ignoring some of the options? If the latter, why?

Reply:
The former. The `FmtSpan` type is actually defined as a wrapper around a `u8`, not an `enum`:

```
pub struct FmtSpan(u8);

impl FmtSpan {
    pub const NEW: FmtSpan = FmtSpan(1 << 0);
    // ...
}
```

https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/format/struct.FmtSpan.html

So these _look_ like `enum` variants, but they're actually constants in the type's namespace. (You can tell something funky is up because they're written in `ALL_CAPS` rather than `TitleCase`.)

In short: this is necessary to make the compiler not complain.